src/shipit_static_analysis/shipit_static_analysis/clang.py

```diff
@@ -251,6 +256,18 @@ def is_third_party(self):
                 return True
         return False
 
+    def is_expanded_macro(self):
+        '''
+        Is the issue only found in an expanded macro ?
+        '''
+        if not self.notes:
+            return False
+
+        return all([
+            CLANG_MACRO_DETECTION.match(note.message) is not None
+            for note in self.notes
+        ])
```

Comment:
Does this mean that we only consider something an expanded macro if *all* of its notes contain the string `expanded from macro`? I would argue that we should omits any issues when its *first* note contains the string (regardless of what all other notes say).

I think that in all examples we've seen until now, both options have the same result, but it's more precise to say "don't complain about an issue in code that was immediately expanded (and may have other notes as well)" than "don't complain about an issue when the entire chain of notes is all about macros".